src/keystore/InMemoryKeystore.ts

```diff
@@ -294,10 +295,14 @@ export default class InMemoryKeystore implements Keystore {
           )
         }
 
+        const keyMaterial = getKeyMaterial(topicData.invitation)
```

Comment:
The changes here currently use the topic key as the input key material, with the topic as a salt. I think we should do one of the following:

1. The source key material for the hkdf (as opposed to the encryption) is per-user, i.e. derived from the private key in the private key bundle. This ensures different users in the topic cannot produce hmac keys for each other. The salt will include the topic, to ensure separation between different topics.

2. If the source key material is derived from the topic key as in the current code, we do not need the topic in the salt, but it needs to include the sender, so that different senders produce different hmac keys. Because all users in the topic know the topic key, they are technically able to forge the hmacs for each other.

(1) is more robust, but (2) is more deniable. We don't have deniability in v2 or v3 anyway, however in v3 it might be more difficult to sync a per-user key, so I lean slightly towards (2). Would love it if @Bren2010 and @neekolas could weigh in here!

Reply:
So, yes. We need some way of scoping it so that the HMAC only verifies correctly from one sender and not the other.

I like 2 for the compatibility with MLS. We could just include the sender's wallet address into the salt or something.